### PR TITLE
feat: Add library books management functionality

### PR DESCRIPTION
**Pull request by Irfan778899**

### CODE REVIEW

File: casecentral/case_central/doctype/book/book.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            book_doc['book_title'] = book.get('title')
+            book_doc['book_subtitle'] = book.get('subtitle')
+            book_doc['author'] = ', '.join(book.get('authors', []))
+            book_doc['category'] = ', '.join(book.get('categories', []))
+            book_doc['publisher'] = book.get('publisher')
+            book_doc['published_date'] = book.get('publishedDate')
+            book_doc['page_count'] = book.get('pageCount')
+            book_doc['preview_link'] = book.get('previewLink')
+            book_doc['description'] = book.get('description')
+            book_doc['book_image'] = book.get('imageLinks')['thumbnail']
```
Comment (mujeerhashmi) [inline casecentral/case_central/doctype/book/book.py:30]: Make sure book.get('imageLinks') is not null

File: casecentral/case_central/doctype/book/book.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+					isbn: frm.doc.isbn
+				},
+				callback: (r) => {
+					var book_details = r.message;
+					// Update the form fields with fetched book details using frm.set_value()
+					Object.keys(book_details).forEach(function(key){
+						frm.set_value(key, book_details[key]);
+					});
+
+					frm.reload_doc();
```
Comment (mujeerhashmi) [inline casecentral/case_central/doctype/book/book.js:28]: Not Required.

**Review by mujeerhashmi — CHANGES_REQUESTED** [review]

File: casecentral/case_central/doctype/book/book.json [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+ "editable_grid": 1,
+ "engine": "InnoDB",
+ "field_order": [
+  "naming_series",
+  "book_title",
+  "purchase_invoice",
+  "column_break_1",
+  "status",
+  "item_code",
+  "item_name",
```
[the remark this answers is not in the document]
Comment (Irfan778899) [reply casecentral/case_central/doctype/book/book.json:18]: Added link field to warehouse

File: casecentral/case_central/doctype/book/book.js (hunk 2 above)
Comment (Irfan778899) [reply casecentral/case_central/doctype/book/book.js]: removed frm.reload_doc();

**Review by mujeerhashmi — APPROVED** [review]

**Pull request merged**